wave-sort.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+using namespace std;
+
+void WaveSort(int arr[], int n)
+{
+    for(int i=1;i<n;i+=2)
+    {
+        if(arr[i] > arr[i+1]  && i<=n-2)
+        {
+            int temp=arr[i];
+            arr[i+1]=arr[i];
+            arr[i]=temp;
+        }
+        if(arr[i] > arr[i-1])
+        {
+            int temp=arr[i];
+            arr[i]=arr[i-1];
+            arr[i-1]=temp;
+
+        }
+    }
+}
+
+int main()
+{
+    cout << "Enter the number of elements: ";
+    int n;
+    cin >> n;
+    int arr[n];

```

Comment:
No VLAs[variable size arrays] please. 
these things are prohibited in C++ and using this here makes your program dependent on language implementation to have a non-standard extension to support this.

it's much better to use some `stl` containers ( like `std::vector`) for such job, or if you are feeling brave then dynamically allocate the memory to the array here ( don't forget to manage it's ownerships and deallocate it if you choose this way )